src/utils/rabbitmq.js

```diff
@@ -1,21 +1,129 @@
 const { Configuration } = require('@schul-cloud/commons');
 const amqp = require('amqplib');
+const logger = require('../logger');
 
 let connection;
 
-const createChannel = async () => {
+const getConnection = async () => {
+	if (!Configuration.get('FEATURE_RABBITMQ_ENABLED') || !Configuration.get('RABBITMQ_URI')) {
+		throw new Error('RabbitMQ is not configured.');
+	}
+
 	if (connection) {
-		const con = await connection; // ensure connection has resolved
-		return con.createChannel();
+		return connection;
 	}
 
-	throw new Error('No RabbitMQ connection available.');
+	try {
+		connection = await amqp.connect(Configuration.get('RABBITMQ_URI'));
+		connection.on('close', () => {
+			logger.warning('RabbitMQ connection was closed.');
+			connection = undefined;
+		});
+		return connection;
+	} catch (err) {
+		connection = undefined;
+		throw err;
```

Comment:
How is thes exception handelt?

Reply:
The exception is bubbled up to be handled where more context is available. In this use case the Channel class catches all RabbitMQ errors and logs them because the messenger sync is done in the background and should not block/stop any user actions.